sips/sip-additional_openid.md

```diff
@@ -0,0 +1,114 @@
+| SIP-Number          |  |
+| ---:                | :--- |
+| Title               | Add Karrier One as a zkLogin OpenID provider |
+| Description         | Add Karrier One as a whitelisted OpenID provider enabled for zkLogin on Sui. |
+| Author              | Andrew Buchanan <abuchanan@karrier.one> |
+| Editor              |  |
+| Type                | Standard |
+| Category            | Core |
+| Created             | 2024-08-06 |
+| Comments-URI        | |
+| Status              | |
+| Requires            | |
+
+
+## Abstract
+
+Currently no providers enabled on Sui support login with a phone number. Phone number logins are a dominant credential option in many sectors and having such a provider for zkLogin is crucial for mass adoption of zkLogin. Here we propose to add Karrier One as an OpenID provider on Sui that enables login with a phone number. This will allow applications in the sui ecosystem to onboard users to Sui seamlessly using simply a phone number.
+
+## Motivation
+
+zkLogin does not support any providers using phone number-based logins.  Many popular applications use phone number based accounts and/or logins today and adding a zkLogin provider that supports phone numbers will enable builders in the sui ecosystem to also use phone number based logins for creating wallets and interacting with dApps.
+
+## Specification
+
+Karrier One is hosting an openid provider built upon https://github.com/openiddict/openiddict-core which implements https://openid.net/specs/openid-connect-core-1_0.html and supports the code & implicit flows. TODO openiddict does support more flows if we want.
+
+
+|             Item          | Endpoint  | Example Content | 
+|-------------------------- |-----------|-----------------|
+| Well known configuration  |    https://accounts.karrier.one/.well-known/openid-configuration       |                 |
+| JWK endpoint              |    https://accounts.karrier.one/.well-known/jwks       |                 |
+| Issuer                    |    https://accounts.karrier.one/   |                 |
+| Authorization link          |   https://accounts.karrier.one/connect/authorize        |                 |
+| Allowed Client IDs |    |     | 
```

Comment:
what client ids are allowed?

Reply:
I've added the allowed client ids.  Currently it's just karrier.one but the system is designed for (and supports) multiple client ids.  I expect we'd want to integrate other projects.